instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/RuleParser.java

```diff
@@ -70,6 +70,14 @@ public void addMetricDefsTo(MetricConfiguration conf, InputStream is, String id)
       // It is essential that the parser exception is made visible to the user.
       // It contains contextual information about any syntax issues found by the parser.
       logger.log(WARNING, exception.toString());
+    } catch (NoClassDefFoundError e) {
+      // snakeyaml uses java.beans.Introspector to get/set properties on an object; if module
+      // java.desktop is not present, trying to parse yaml will result in a NoClassDefFoundError
+      logger.log(
+          WARNING,
+          "Failed to parse YAML rules from {0}: {1}. "
+              + "If you're using a custom JRE distro built using jlink, ensure that the java.desktop module is included.",
+          new Object[] {id, e.toString()});
```

Comment:
thoughts on using jackson yaml? cc @jack-berg

also snakeyaml is not very responsive at dealing with CVE reports so [our CVE check](https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/3907684368/jobs/6677155250) has been failing for a while now due to

```
One or more dependencies were identified with known vulnerabilities in javaagent:

snakeyaml-1.33.jar (pkg:maven/org.yaml/snakeyaml@1.33, cpe:2.3:a:snakeyaml_project:snakeyaml:1.33:*:*:*:*:*:*:*, cpe:2.3:a:yaml_project:yaml:1.33:*:*:*:*:*:*:*) : CVE-2022-1471, CVE-2022-3064, CVE-2021-4235
```

Reply:
I think Jackson would be safer as well.

Reply:
https://github.com/FasterXML/jackson-dataformats-text/tree/2.15/yaml seems to use snakeyaml under the hood. Besides [snakeyaml](https://bitbucket.org/snakeyaml/snakeyaml/src/master/) there is also [snakeyaml engine](https://bitbucket.org/snakeyaml/snakeyaml-engine/src/master/) that claims to offer safe parsing of yaml documents.

Reply:
Let me try snakeyaml-engine then.

Reply:
Done!